### ISSUE TITLE
skiplines in CSVParser not working when useColumnNames is true

### ISSUE THREAD

**Issue by mariosanchez23** [amcharts/amcharts4#1212]
There is no way to parse a csv file which first row/s are not useful when we still have the column names row after the second or later rows. 

When you set skip lines and useColumnNames together, you never skip any line and take the column names from the first row . Which may is by design, but just a simple change will improve in a way that you could skip un-useful lines before the row with data names. 

Looking at the CSVParser.ts I think that changing the order of the "ifs" would work. 

So instead of this: 

            if (this.options.useColumnNames) {
			cols = data.shift();
                        ... 
            ....
		// Skip rows
		for (i = 0; i < this.options.skipRows; i++) {
			data.shift();
		}

Change to this: 

		// Skip rows
		for (i = 0; i < this.options.skipRows; i++) {
			data.shift();
		}

            if (this.options.useColumnNames) {
			cols = data.shift();

This will allow to use skip lines to let you position on the line with column names. 

If there is any workaround please tell me. I am not allowed to touch the source file, and adding column names manually is not an option. 
In amcharts V3 I solved it using the DataSource plugin which was a little more flexible and allow me to position in the unparsed string.

**Comment by martynasma** [comment]
Thanks!

I totally agree it would make sense that `skipRows` should kick in first, then `useColumnNames` would take the first row from what's left.

Also, since you asked about ability to manipulate data before parsing starts, there is one: `unparsedData` adapter:

```
chart.dataSource.adapter.add("unparsedData", function(data) {
  // do something to data
  // ...
  return data;
});
```

In any case, I think we'll need to make the switch of the skipRows doing its thing first.

**Comment by martynasma** [comment]
Fixed in [4.4.1](https://github.com/amcharts/amcharts4/releases/tag/4.4.1).

## [4.4.1] - 2019-04-22

### Changed
- Now when using `CSVParsing` for parsing external data, `skipRows` will kick in first, removing X rows from the beginning of data, then `useColumnNames` will use the first row of what's left.

### Fixed
- JSON: Specifying language locale as string (i.e. "fr_FR") now works in JS setups, provided appropriate language file is loaded.
- Grab/Grabbing cursor was not appearing on Scrollbars on Firefox.
- `MapChart` used to pan itself when it was zoomed in, `tapToActivate` enabled, and was tapped.
- Hovering mouse cursor over an `XYChart` while it was initializing could sometimes result in a JS error.
- Issue with `DateAxis` and `Series` not having dates at certain data items fixed. It was cousing problems with multiple date axes mostly.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
